### ISSUE TITLE
Optionally support thick binaries for compile

### ISSUE THREAD

**Issue by KamiQuasi** [denoland/deno#10041]
Given the following files
```javascript
// test.ts
class TestClass {
    stuff = '';
    constructor() {
        this.stuff = 'foo';
    }
}
```

and 

```javascript
// emit.ts
const fileName = "test.ts";
const { files, diagnostics } = await Deno.emit(fileName, {
  check: true,
  bundle: "esm",
  compilerOptions: {
    lib: ["es6", "dom"],
    module: "es6",
    target: "es6",
  },
});
if (diagnostics.length) {
  console.warn(Deno.formatDiagnostics(diagnostics));
}
```

When I run the following commands the results follow:
* No error - `deno run --unstable --allow-read emit.ts` - No Error
* Error - `deno compile --output=emitter_test --unstable --allow-read emit.ts && ./emitter_test`

Error is:

```
error: TypeError: invalid op id: expected type `rusty_v8::data::Integer`, got `rusty_v8::data::Value`
    at opAsync (deno:core/core.js:249:10)
    at Object.jsonOpAsync (deno:core/core.js:364:12)
    at opEmit (deno:runtime/js/40_compiler_api.js:42:17)
    at Object.emit (deno:runtime/js/40_compiler_api.js:83:12)
    at file://$deno$/bundle.js:2:45
```

**Comment by lucacasonato** [comment]
`Deno.emit` intentionally does not work in compiled binaries to reduce the `--lite` mode size of standalone binaries. The error is not great, but it not working is correct.

**Comment by KamiQuasi** [comment]
@lucacasonato I don't have a super strong argument for this. The use case this came up in was a web server that hosts TS files that are proxied to JS on-demand but cached. There are workarounds, but the error was definitely unexpected as it worked under `deno run` just the way I wanted it to. The compile came from trying to containerize the script, which is not completely necessary.

**Comment by kitsonk** [comment]
This is effectively suggesting that we support "thick" binaries for compile. I have changed the title to reflect that.

**Comment by littledivy** [comment]
Aren't "thick"/main binaries already the default for `deno compile`?

**Comment by iugo** [comment]
> Aren't "thick"/main binaries already the default for `deno compile`?

deno compile still needs the `--lite` option. (It is not included in 1.10)

**Comment by crowlKats** [comment]
@iugo lite was removed
